distribution/src/bin/elasticsearch

```diff
@@ -17,15 +17,19 @@ source "`dirname "$0"`"/elasticsearch-env
 
 CHECK_KEYSTORE=true
 DAEMONIZE=false
-for option in "$@"; do
-  case "$option" in
-    -h|--help|-V|--version)
-      CHECK_KEYSTORE=false
-      ;;
-    -d|--daemonize)
-      DAEMONIZE=true
-      ;;
-  esac
+ENROLL_TO_CLUSTER=false
+# Store original arg array as we will be shifting through it below
+ARG_LIST=($@)
+while [ $# -gt 0 ]; do
+   if [[ $1 == "--enrollment-token" ]]; then
+     ENROLL_TO_CLUSTER=true
+     ENROLLMENT_TOKEN="$2"
+   elif [[ $1 == "-h" || $1 == "--help" || $1 == "-V" || $1 == "--version" ]]; then
+     CHECK_KEYSTORE=false
+   elif [[ $1 == "-d" || $1 == "--daemonize" ]]; then
```

Comment:
It's a little safer to quote the args here, in case someone passed an argument with a space in it.

```suggestion
   if [[ "$1" == "--enrollment-token" ]]; then
     ENROLL_TO_CLUSTER=true
     ENROLLMENT_TOKEN="$2"
   elif [[ "$1" == "-h" || "$1" == "--help" || "$1" == "-V" || "$1" == "--version" ]]; then
     CHECK_KEYSTORE=false
   elif [[ "$1" == "-d" || "$1" == "--daemonize" ]]; then
```

Reply:
Huh. TIL that it's not necessary if you're using `[[` - https://stackoverflow.com/a/3427931/16173444